nautobot/docs/development/jobs/job-patterns.md

```diff
@@ -0,0 +1,519 @@
+
+        def run(self):
+            config = self.load_yaml("defaults.yaml")
+            self.logger.info("Loaded configuration with %d items.", len(config))
+
+    register_jobs(LoadConfig)
+    ```
+
+Unlike uploaded files, static files are typically bundled with the Job and version-controlled. This makes them useful for Jobs that rely on standard schemas or reusable templates.
+
+## Reference Implementations
+
+The examples below demonstrate fully working Jobs from the Nautobot ecosystem. They're meant to serve as inspiration and guidance when building your own Jobs—especially if you're not sure where to start or want to see how others have structured their code.
+
+Some Jobs are simple and focused (like validating devices), while others combine multiple input types and outputs. Use these as starting points, or to understand common patterns in action.
+
+### Device Validation
+
+This Job performs checks to ensure all active devices in Nautobot meet certain physical connection criteria. It does not require any user input.
+
+This Job is useful when enforcing operational standards in environments where minimum connectivity (e.g., console and dual power) is expected. It demonstrates how to iterate over Nautobot models, access related objects, and use structured logging with `extra`. Since it defines no variables, it can be run immediately by any user with Job permissions.
+
+!!! note "Requirements"
+    - Devices must exist in the database with `status="Active"`
+    - Devices must have at least two connected PowerPorts for validation to pass
+    - ConsolePorts should have active connections
+    - The `Status` model must include an `"Active"` value
+
+<!-- pyml disable-num-lines 10 proper-names -->
+!!! example "Device Validation Job"
+    ```py
+    from nautobot.apps.jobs import Job, register_jobs
+    from nautobot.dcim.models import ConsolePort, Device, PowerPort
+    from nautobot.extras.models import Status
+
+    class DeviceConnectionsReport(Job):
+        description = "Validate the minimum physical connections for each device"
+
+        def test_console_connection(self):
+            STATUS_ACTIVE = Status.objects.get(name='Active')
+
+            # Check that every console port for every active device has a connection defined.
+            for console_port in ConsolePort.objects.select_related('device').filter(device__status=STATUS_ACTIVE):
+                if console_port.connected_endpoint is None:
+                    self.logger.error(
+                        "No console connection defined for %s",
+                        console_port.name,
+                        extra={"object": console_port.device},
+                    )
+                else:
+                    self.logger.info(
+                        "Console port %s has a connection defined",
+                        console_port.name,
+                        extra={"object": console_port.device},
+                    )
```

Comment:
*might* be nice to use `logger.failure` and `logger.success` instead in this job here and below?